internal/driver/glfw/window_desktop.go

```diff
@@ -261,6 +261,26 @@ func (w *window) fitContent() {
 	}
 }
 
+// getMonitorScale returns the scale factor for a given monitor, handling platform-specific cases
+func getMonitorScale(monitor *glfw.Monitor) float32 {
+	widthMm, heightMm := monitor.GetPhysicalSize()
+	if runtime.GOOS == "linux" && widthMm == 60 && heightMm == 60 { // Steam Deck incorrectly reports 6cm square!
+		return 1.0
+	}
+	widthPx := monitor.GetVideoMode().Width
+	return calculateDetectedScale(widthMm, widthPx)
+}
+
+// getScaledMonitorSize returns the monitor dimensions adjusted for scaling
+func getScaledMonitorSize(monitor *glfw.Monitor) (int, int) {
```

Comment:
Thanks this is cleaner now, but I do think that the return from this should be `fyne.Size` because (if I understand it correctly) that is the coordinate system being used for the return.